src/Synapse/Page/Bundle/Resources/config/services/Page/domain.xml

```diff
@@ -45,6 +45,7 @@
             </argument>
             <call method="setEventDispatcher"><argument type="service" id="event_dispatcher" /></call>
             <call method="setValidator"><argument type="service" id="validator" /></call>
+            <call method="setRepository"><argument type="service" id="synapse.page.repository" /></call>
```

Comment:
If you're purpose is to check if a page already exists into persistence, you have to load it. You should use "loader" service for this, which already have a `retrieveByPath()` method.
On the other hand, if you want to *validate* any page doesnt exists, you could use a custom validation constraint, and register it into proper `validation.yml` file.